examples/ResNet/ResNet.cpp

```diff
@@ -18,7 +18,7 @@
 
 #include "common/Log.h"
 
-ResNet::ResNet() {
+ResNet::ResNet(utils::ExampleOptions options) {
```

Comment:
How about adding a basic class like ExampleBase and move L.22 - L.30 to the class, and move utils::ImagePreprocessOptions, imagePath, weightsPath to each subclass so that ImagePreprocessOptions/ExampleOptions is unnecessary?